docs/source/user_guide/difference.rst

```diff
@@ -115,6 +115,17 @@ This may affect the dtype of its output, depending on the values of the "scalar"
   dtype('float64')
 
 
+Matrix type
+-----------
+
+CuPy does not have the ``matrix`` subclass of :class:`cupy.ndarray`.
+:class:`numpy.matrix` is discouraged and will be removed in the future.
+
+Because `sparse ndarray class <https://github.com/scipy/scipy/issues/8162>`_ has not been introduced to SciPy,
+the sparse matrix classes in :mod:`cupyx.scipy.sparse` often behave like :class:`numpy.matrix`, too.
+The classes in CuPy, however, return :class:`cupy.ndarray`\s when dense matrices are computed from sparse matrices.
```

Comment:
```suggestion
SciPy returns :class:`numpy.matrix` when dense matrices are computed from sparse matrices (e.g., ``coo_matrix + ndarray``). However, CuPy returns :class:`cupy.ndarray` for such operations.

There is no plan to provide :class:`numpy.matrix` equivalent in CuPy.
This is because the use of :class:`numpy.matrix` is no longer recommended since NumPy 1.15.
```

* I reversed the sentence to try to improve the understandability.
* Mentioning https://github.com/scipy/scipy/issues/8162 is interesting (I didn't know this is discussed in SciPy...!) but I felt a bit out of context of this docs.
* Changed the tone of the sentence, as the docs say `The class may be removed in the future.` and not `will be removed` 😃 

Evidence: https://github.com/numpy/numpy/pull/10142, https://numpy.org/doc/1.15/reference/generated/numpy.matrix.html#numpy.matrix